### PR TITLE
Upgrade addon-resizer to 1.8.3

### LINKED ISSUE #1201: kube-state-metrics pod becomes pending after adding a node from metalk8s-ui
**Component**:

salt, metalk8s-ui

**What happened**:

After adding a cp node from metalk8s-ui, the kube-state-metrics pod goes to Pending state

**What was expected**:

kube-state-metrics should remain Running and should be scheduled on all nodes of the cluster

**Steps to reproduce**

DO: deploy metalk8s bootstrap node
OBSERVE: kube-state-metrics pod is in RUNNING state
DO: add and deploy a node from metalk8s-ui
EXPECT: kube-state-metrics is scheduled on the new node and is in running state
OBSERVE: kube-state-metrics is **not** scheduled on the new node and goes to pending state

**Resolution proposal** (optional):

### PR DESCRIPTION
**Pull request by TeddyAndrieux**
**Component**:

'salt', 'deployment', 'monitoring'

**Context**: 

addon_resizer update `kube-state-metrics` deployment to resize the `kube-state-metrics` container according to the size fo the cluster, in version 1.0 due to a bug (https://github.com/kubernetes/autoscaler/issues/188) tolerations get removed.

**Summary**:

Upgrade addon_resizer to 1.8.3 to fix this issue

**Acceptance criteria**: 

Deployment `kube-state-metrics` tolerations not get removed

---

Fixes: #1201

### CODE REVIEW

**Comment by TeddyAndrieux** [comment]
@bert-e reset

**Comment by TeddyAndrieux** [comment]
@bert-e approve

**Review by sayf-eddine-scality — APPROVED** [review]
LGTM tested

File: salt/metalk8s/addons/monitoring/kube-state-metrics/upstream.sls [hunk 1]
```diff
@@ -219,11 +219,11 @@ spec:
         name: kube-state-metrics
         resources:
           limits:
-            cpu: 102m
```
Comment (slaperche-scality) [inline salt/metalk8s/addons/monitoring/kube-state-metrics/upstream.sls:222]: This part should be reverted in a dedicated commit, for "historical" purpose (you can probably use `git revert`).

**Review by slaperche-scality — COMMENTED** [review]
LGTM overall, except for the revert
